src/core/component.js

```diff
@@ -121,7 +122,17 @@ Component.prototype = {
       return error(typeName + ' is not a valid type.');
     }
 
-    return objectParse(value);
+    value = objectParse(value);
+
+    if (value === null || typeof value !== 'object') { return value; }
+
+    Object.keys(value).forEach(parseProperty)
+    return value;
+
+    function parseProperty (key) {
+      if (!schema[key] || !schema[key].parse) { return; }
```

Comment:
If the key is there, then it should always have a `parse` method as given by the schema processor.